### PR TITLE
various fixes for event addition/removal

### LINKED ISSUE #42: EventEmitter memory leak
I get the following error in console: 

> (node) warning: possible EventEmitter memory leak detected. 11 listeners added. Use emitter.setMaxListeners() to increase limit.

Not sure what causes it, but it seems to be related to dynamic data updates. It can be seen in this example: [CodePen](https://codepen.io/swiperii/pen/wyPEBW?editors=0010). Click "Add data" and watch console.

### PR DESCRIPTION
**Pull request by nicolaskruchten**
Fixes #42

### CODE REVIEW

File: src/__tests__/react-plotly.test.js [hunk 1]
```diff
@@ -135,32 +135,6 @@ describe('<Plotly/>', () => {
           .catch(err => done.fail(err));
       });
 
-      test('clear event handlers on newPlot', done => {
```
Comment (nicolaskruchten) [inline src/__tests__/react-plotly.test.js]: this test was way too implementation-specific and now depends on the Plotly.js version.

File: src/factory.js [hunk 2]
```diff
@@ -97,30 +95,30 @@ export default function plotComponentFactory(Plotly) {
...
               config: nextProps.config,
               frames: nextProps.frames,
             });
           }
         })
         .then(() => this.syncEventHandlers(nextProps))
         .then(() => this.syncWindowResize(nextProps))
-        .then(this.attachUpdateEvents)
+        .then(() => {
+          if (!hasReactAPIMethod) this.attachUpdateEvents();
```
Comment (nicolaskruchten) [inline src/factory.js]: this is the main fix

**Comment by VeraZab** [comment]
this looks good 💃

**Comment by nicolaskruchten** [comment]
Thanks! I'll do a release after I've eaten :)

**Pull request merged**